src/app/modules/form/form.service.ts

```diff
@@ -274,6 +275,50 @@ export const checkFormSubmissionLimitAndDeactivateForm = (
   })
 }
 
+/**
+ * Verify that if the form is a single submission per submitterId form, the submitterId has not responded
+ * @param form the form to check for
+ * @param submitterId submitterId of current submitter
+ * @returns ok(true) if the form is not a single submission per submitterId form or the submitterId has not responded before
+ * @returns ok(false) if the form is a single submission per submitterId form and submitterId has already responded
+ * @returns err(ApplicationError) if the submitterId is not found
+ * @returns err(PossibleDatabaseError) if an error occurred while querying the database to check for previous responses by same submitterId
+ */
+export const checkHasSingleSubmissionValidationFailure = (
+  form: PublicFormDto,
+  submitterId: string,
+): ResultAsync<boolean, ApplicationError | PossibleDatabaseError> => {
+  const logMeta = {
+    action: 'checkHasSingleSubmissionValidationFailure',
+    formId: form._id,
+  }
+  if (form.isSingleSubmission) {
+    if (!submitterId) {
+      return errAsync(
+        new ApplicationError(
+          'Cannot find submitterId which is required for single submission per submitterId form',
+        ),
+      )
+    }
+    const submissionWithSameSubmitterIdExists = SubmissionModel.exists({
+      form: form._id,
+      submitterId,
+    }).then((result) => result != null)
```

Comment:
typo on `!==`? Since result cannot be `undefined`

Reply:
I was thinking if `!=` might be better to be used here

here i want the expression to evaluate to `false` if the result is `undefined` or `null` 

while there isnt a chance of the result being `undefined`, it might be more defensive to use `!=` as it enforces the above intent. 
![image](https://github.com/opengovsg/FormSG/assets/55353265/8ea4c166-5748-425c-a250-b0382052139a)

what are your thoughts on this?